snapcraft/storeapi/_upload.py

```diff
@@ -318,6 +318,30 @@ def get_post_files(metadata=None):
     return files
 
 
+def is_scan_completed(response):
+    """Return True if the response indicates the scan process completed."""
+    if response is None:
+        # To cope with spurious connection failures lacking a proper response:
+        # either we'll retry and succeed or we fail for all retries and report
+        # an error.
+        return False
+    if response.ok:
+        return response.json().get('completed', False)
+    return False
+
+
+def get_scan_status(session, url):
```

Comment:
I think I would name these helper funcions with a leading underscore, to make it clear that they are called only from higher level functions.